lib/logasm/preprocessors/strategies/mask_strategy.js

```diff
@@ -0,0 +1,59 @@
+const { isObject } = require('../../object_helpers');
+
+const WILDCARD_SYMBOL = '~';
+const MASKED_VALUE = '*****';
+
+class MaskStrategy {
+  constructor(whitelistedFields) {
+    this.whitelistedFields = whitelistedFields;
+  }
+
+  process(data, path) {
+    if (path == null) { path = []; }
+    
+    if (this._isWhitelistedField(path)) {
+      return this._processData(data, path)
+    } else {
+      return MASKED_VALUE;
+    }
+  }
+
+  _isWhitelistedField(path) {
```

Comment:
should we cache the result? Btw, do you have the performance results compared to ruby? Even though the performance did not degrade I assume, it would be cool to see how both of the libraries compare to each other

Reply:
Could cache for better performance yeah. What do you think of using this https://github.com/groupon/node-cached Has a similar getset function as we're using in ruby logasm.

Reply:
I guess it can stay for now